### PR TITLE
Auto-load `processing_class` in CPO/ORPO trainers when omitted

### PR DESCRIPTION
**Pull request by DaoyuanLi2816**
## What does this PR do?

`CPOTrainer` and `ORPOTrainer` (under `trl.experimental`) document `processing_class` as *optional* and default it to `None` in their signatures, but they raise instead of handling that case:

```python
if processing_class is None:
    raise ValueError("processing_class must be specified to tokenize a CPO dataset.")
```

So the documented-optional usage — passing just a model — fails:

```python
CPOTrainer(model="Qwen/Qwen2.5-0.5B", args=cfg, train_dataset=ds)  # ValueError
```

This is inconsistent with the non-experimental preference trainers, which auto-load `processing_class` from the model when it is omitted:

- `RewardTrainer` → `AutoTokenizer.from_pretrained(get_config_model_id(model.config))`
- `DPOTrainer` → `AutoProcessor.from_pretrained(get_config_model_id(model.config))`

This PR makes CPO/ORPO mirror `RewardTrainer` (both are text-only preference trainers), so the `*optional*` docstring is honored:

```python
if processing_class is None:
    processing_class = AutoTokenizer.from_pretrained(get_config_model_id(model.config))
```

By this point the model has already been instantiated from a string a few lines above, so `model.config` is available — the same ordering `RewardTrainer` uses. The trainer also sets `pad_token` from `eos_token` right afterwards, so an auto-loaded tokenizer flows through the rest unchanged.

A regression test is added for each trainer: construct it without `processing_class`, assert one is loaded, and run a few training steps.

## Before submitting

- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request)?
- [ ] Was this discussed/approved via a GitHub issue? _(N/A — small consistency fix aligning the experimental CPO/ORPO trainers with `DPOTrainer`/`RewardTrainer`.)_
- [ ] Did you make sure to update the documentation with your changes? _(N/A — the `processing_class` docstring already says *optional*; this makes the behavior match it.)_
- [x] Did you write any new necessary tests? _(Regression test added for each trainer.)_

---

> [!NOTE]
> **Low Risk**
> Initializer-only behavior change with the same pattern used elsewhere in TRL; covered by new regression tests and no change to loss or training logic.
> 
> **Overview**
> **BCOTrainer**, **CPOTrainer**, and **ORPOTrainer** now honor the documented optional `processing_class`: when it is omitted, they load a tokenizer with `AutoTokenizer.from_pretrained(get_config_model_id(model.config), …)` instead of raising `ValueError`, matching **RewardTrainer** / **DPOTrainer**.
> 
> Regression tests construct each trainer without `processing_class`, assert one is set, and run a short training step.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 21ace67140d4c6205b7fbad691975d5c16365e4f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by qgallouedec** [comment]
Thanks @DaoyuanLi2816, nice catch.

One thing before we merge: this same `if processing_class is None: raise ValueError(...)` block (followed by the identical `max_length` default) also lives in `BCOTrainer` (`trl/experimental/bco/bco_trainer.py:572`), the third sibling offline preference trainer, which likewise documents `processing_class` as optional. Right now the PR fixes CPO/ORPO but leaves BCO, so it trades one inconsistency for another. Could you apply the same auto-load + a matching regression test to BCO? (`online_dpo` also raises, but it's an online/generation trainer; leave that one.)

After that this is good to go.

**Comment by DaoyuanLi2816** [comment]
Done 👍 Applied the same fix to `BCOTrainer`: when `processing_class` is `None` it now auto-loads the tokenizer from the model (`AutoTokenizer.from_pretrained(get_config_model_id(model.config))`), mirroring `RewardTrainer`/`KTOTrainer`, and added a matching `test_train_processing_class_autoloaded` regression test. Left `online_dpo` untouched as you suggested.

Locally the new BCO test passes and `ruff check` / `ruff format` are clean.

File: trl/experimental/cpo/cpo_trainer.py [hunk 1]
```diff
@@ -258,7 +264,7 @@ def make_inputs_require_grad(module, input, output):
             self.pad_token_id = model.config.pad_token_id
 
         if processing_class is None:
-            raise ValueError("processing_class must be specified to tokenize a CPO dataset.")
+            processing_class = AutoTokenizer.from_pretrained(get_config_model_id(model.config))
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/experimental/cpo/cpo_trainer.py]: right, can you just add `, trust_remote_code=args.trust_remote_code`

**Comment by DaoyuanLi2816** [comment]
Done, added it to the auto-load in CPO/ORPO/BCO.

Small note: these configs (`_BaseConfig` → `transformers.TrainingArguments`) don't expose a `trust_remote_code` field, so `args.trust_remote_code` would `AttributeError`. I sourced it from `model_init_kwargs` instead — i.e. the same kwargs the model itself is loaded with:

```python
processing_class = AutoTokenizer.from_pretrained(
    get_config_model_id(model.config),
    trust_remote_code=model_init_kwargs.get("trust_remote_code", False),
)
```

So a remote-code checkpoint passed as `CPOTrainer(model="...", args=CPOConfig(model_init_kwargs={"trust_remote_code": True}))` now loads its tokenizer with the same flag. Happy to switch to a different source if you'd prefer.

**Comment by qgallouedec** [comment]
BCO CPO and ORPO all have a `trust_remote_code` arg now:

https://github.com/DaoyuanLi2816/trl/blob/cf5d0d24443fd0d32bc9b46a291c57a87328ff7b/trl/experimental/bco/bco_config.py#L142-L148

**Comment by DaoyuanLi2816** [comment]
Ah, you're right — those `trust_remote_code` config fields landed on `main` after I branched, so my earlier base didn't have them (hence the `model_init_kwargs` workaround). Now that the branch is updated, I've switched all three to `trust_remote_code=args.trust_remote_code`. Thanks!

**Review by qgallouedec — APPROVED** [review]
